src/coreclr/src/pal/src/map/virtual.cpp

```diff
@@ -2284,7 +2289,11 @@ int32_t ExecutableMemoryAllocator::GenerateRandomStartOffset()
     // This code is similar to what coreclr runtime does on Windows.
     // It generates a random number of pages to skip between 0...MaxStartPageOffset.
     srandom(time(NULL));
-    pageCount = (int32_t)(MaxStartPageOffset * (int64_t)random() / RAND_MAX);
+#if HAVE_ARC4RANDOM
+    pageCount = (int32_t)(MaxStartPageOffset * (int64_t)arc4random() / OFFSET_RAND_MAX);
```

Comment:
Maybe use the arc4random_uniform flavor so that we don't need to do the multiplication and division:
```suggestion
    pageCount = (int32_t)arc4random_uniform(MaxStartPageOffset);
```

Reply:
Thanks. also updated the introspection and macro name.

Reply:
@janvorli, should I remove OFFSET_RAND_MAX, as both Illumos and Solaris will take this branch?

Reply:
Yes, please remove that.

Reply:
Removed in the latest commit.